test/transform.spec.ts

```diff
@@ -0,0 +1,221 @@
+    it("generates transform w/ scale", function () {
+      const point = p.point(1, 1)
+      const trans = t.scale(1, 1)
+
+      expect(trans.scale.value).to.eql(point)
+      expectTransShape(trans)
+    })
+  })
+
+  describe("scalePoint(s: Point)", function() {
+    it("generates transform w/ point scale", function () {
+      const point = p.point(1, 1)
+      const trans = t.scalePoint(point)
+
+      expect(trans.scale.value).to.eql(point)
+      expectTransShape(trans)
+    })
+  })
+
+  describe("scaleAround(x: number, y: number, sx: number, sy: number)", function() {
+    it("generates transform w/ scale", function () {
+      const point = p.point(1, 2)
+      const scale = p.point(3, 4)
+
+      const trans = t.scaleAround(1, 2, 3, 4)
+
+      expect(trans.scale.value).to.eql(scale)
+      expect(trans.scale.anchor).to.eql(point)
+      expectTransShape(trans)
+    })
+  })
+  describe("scaleAroundPoint(a: Point, s: Point)", function() {
+    it("generates transform w/ point scale", function () {
+      const point = p.point(1, 2)
+      const scale = p.point(3, 4)
+
+      const trans = t.scaleAroundPoint(point, scale)
+      
+      expect(trans.scale.value).to.eql(scale)
+      expect(trans.scale.anchor).to.equal(point)
+      expectTransShape(trans)
+    })
+  })
+
+  describe("opacity(o: number)", function () {
+    it("generates transform w/ opacity", function () {
+      const opacity = 1
+
+      const trans = t.opacity(opacity)
+
+      expect(trans.opacity).to.equal(opacity)
+      expectTransShape(trans)
+    })
+  })
+
+  describe("compose(t1: Transform, t2: Transform)", function() {
+    it("Can compose two translations", function() {
+
+      const transform1 = t.transform(p.point(2, 3), a.unit(90), a.unit(p.point(5, 5)), 0.5)
+      const transform2 = t.transform(p.point(3, 2), a.unit(90), a.unit(p.point(2, 2)), 0.5)
+      
+      const expectedTransform = {
+        translate: { x: 5, y: 5 },
+        rotate: { anchor: { x: 0, y: 0 }, value: 180 },
+        scale: { anchor: { x: 0, y: 0 }, value: { x: 10, y: 10 } },
+        opacity: 0.25
+      }
+
+      const trans = t.compose(transform1, transform2)
+
+      expect(trans).to.deep.eq(expectedTransform)
+    })
+  })
+  describe("composeAll(ts: Array<Transform>)", function() {
+    it("Can compose three translations", function () {
+      
+      const transform1 = t.transform(p.point(2, 3), a.unit(90), a.unit(p.point(5, 5)), 0.5)
+      const transform2 = t.transform(p.point(3, 2), a.unit(90), a.unit(p.point(2, 2)), 0.5)
+      const transform3 = t.transform(p.point(5, 5), a.unit(90), a.unit(p.point(2, 2)), 2)
+
+      const expectedTransform = {
+        translate: { x: 10, y: 10 },
+        rotate: { anchor: { x: 0, y: 0 }, value: 270 },
+        scale: { anchor: { x: 0, y: 0 }, value: { x: 20, y: 20 } },
+        opacity: 0.5
+      }
+
+      const trans = t.composeAll([transform1, transform2, transform3])
+
+      expect(trans).to.deep.eq(expectedTransform)
```

Comment:
to.deep.eq i presume means deep equal. i.e all it's children properties must be exactly the same kinda like `cloneNode(true);` ?

Reply:
Correct, so normal eq:

```

const a = {foo: {bar: "lol"}}
const b = {foo: {bar: "Thank you for the music"}}

expect(a).to.eq(b) // true
expect(a).to.deep.eq(b) // error

```

ish